rq/job.py

```diff
@@ -609,9 +606,12 @@ def cleanup(self, ttl=None, pipeline=None, remove_from_queue=True):
             self.delete(pipeline=pipeline, remove_from_queue=remove_from_queue)
         elif not ttl:
             return
-        elif ttl > 0:
+        else:
             connection = pipeline if pipeline is not None else self.connection
-            connection.expire(self.key, ttl)
+            if ttl > 0:
+                connection.expire(self.key, ttl)
+            else:
+                connection.persist(self.key)
```

Comment:
I think reverting it back to what it was is the better option (don't do anything, because this already means that the key won't be expired).